src/borg/key.py

```diff
@@ -258,8 +254,8 @@ def id_hash(self, data):
         return sha256(data).digest()
 
     def encrypt(self, chunk):
-        chunk = self.compress(chunk)
```

Comment:
IIRC we discussed this many months ago and whether we maybe want to tell the repository which is data and which is metadata but decided against it. Since this is an API that is exclusively used within key.py it has no big impact either way (see diff size of the commit), apart from making the code a bit simpler.